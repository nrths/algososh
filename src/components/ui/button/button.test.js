import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Buttons tests', () => {

    it('button with text', () => {
        const button = renderer.create(<Button text="foo" />).toJSON()
        expect(button).toMatchSnapshot()
    })

    it('button without text', () => {
        const button = renderer.create(<Button />).toJSON()
        expect(button).toMatchSnapshot()
    })

    it('button disabled', () => {
        const button = renderer.create(<Button disabled />).toJSON()
        expect(button).toMatchSnapshot()
    })

    it('button load', () => {
        const button = renderer.create(<Button isLoader />).toJSON()
        expect(button).toMatchSnapshot()
    })

    it('button cb works correctly', () => {
        window.alert = jest.fn()
        render(<Button text='foo' onClick={alert('test')} />)
        const button = screen.getByText('foo')
        fireEvent.click(button)
        expect(window.alert).toHaveBeenCalledWith('test')
      })
})