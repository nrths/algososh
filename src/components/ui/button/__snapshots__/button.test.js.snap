// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons tests button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Buttons tests button load 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Buttons tests button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    foo
  </p>
</button>
`;

exports[`Buttons tests button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
